packages/compiler-cli/src/ngtsc/imports/src/default.ts

```diff
@@ -89,65 +93,25 @@ export class DefaultImportTracker {
-      };
-    };
-  }
+    return context => {
+      let clausesToPreserve: Set<ts.Declaration>|null = null;
 
-  /**
-   * Process a `ts.SourceFile` and replace any `ts.ImportDeclaration`s.
-   */
-  private transformSourceFile(sf: ts.SourceFile): ts.SourceFile {
-    const originalSf = ts.getOriginalNode(sf) as ts.SourceFile;
-    // Take a fast path if no import declarations need to be preserved in the file.
-    if (!this.sourceFileToUsedImports.has(originalSf)) {
-      return sf;
-    }
-
-    // There are declarations that need to be preserved.
-    const importsToPreserve = this.sourceFileToUsedImports.get(originalSf)!;
-
-    // Generate a new statement list which preserves any imports present in `importsToPreserve`.
-    const statements = sf.statements.map(stmt => {
-      if (ts.isImportDeclaration(stmt) && importsToPreserve.has(stmt)) {
-        // Preserving an import that's marked as unreferenced (type-only) is tricky in TypeScript.
-        //
-        // Various approaches have been tried, with mixed success:
-        //
-        // 1. Using `ts.updateImportDeclaration` does not cause the import to be retained.
-        //
-        // 2. Using `ts.factory.createImportDeclaration` with the same `ts.ImportClause` causes the
-        //    import to correctly be retained, but when emitting CommonJS module format code,
-        //    references to the imported value will not match the import variable.
-        //
-        // 3. Emitting "import * as" imports instead generates the correct import variable, but
-        //    references are missing the ".default" access. This happens to work for tsickle code
-        //    with goog.module transformations as tsickle strips the ".default" anyway.
-        //
-        // 4. It's possible to trick TypeScript by setting `ts.NodeFlag.Synthesized` on the import
-        //    declaration. This causes the import to be correctly retained and generated, but can
-        //    violate invariants elsewhere in the compiler and cause crashes.
-        //
-        // 5. Using `ts.getMutableClone` seems to correctly preserve the import and correctly
-        //    generate references to the import variable across all module types.
-        //
-        // Therefore, option 5 is the one used here. It seems to be implemented as the correct way
-        // to perform option 4, which preserves all the compiler's invariants.
-        //
-        // TODO(alxhub): discuss with the TypeScript team and determine if there's a better way to
-        // deal with this issue.
-        // tslint:disable-next-line: ban
-        stmt = ts.getMutableClone(stmt);
-      }
-      return stmt;
-    });
+      return sourceFile => {
+        const clausesForFile = this.sourceFileToUsedImports.get(sourceFile.fileName);
 
-    // Save memory - there's no need to keep these around once the transform has run for the given
-    // file.
-    this.sourceFileToUsedImports.delete(originalSf);
+        if (clausesForFile) {
```

Comment:
In the compiler code, explicitly check for `undefined`
```suggestion
        if (clausesForFile !== undefined) {
```